mod-source-record-manager-server/src/main/java/org/folio/dao/JournalRecordDaoImpl.java

```diff
@@ -178,82 +179,38 @@ public Future<String> save(JournalRecord journalRecord, String tenantId) {
       .onFailure(e -> LOGGER.warn("save:: Error saving JournalRecord entity", e));
   }
 
-  @Override
-  public Future<List<RowSet<Row>>> saveBatch(Collection<JournalRecord> journalRecords, String tenantId) {
-    LOGGER.debug("saveBatch:: Starting batch save of {} JournalRecord entities", journalRecords.size());
-    Promise<List<RowSet<Row>>> promise = Promise.promise();
+  public Future<Void> saveBatch(Collection<JournalRecord> journalRecords, String tenantId) {
+    LOGGER.info("saveBatch:: Saving {} journal records", journalRecords.size());
 
     try {
-      // Group records by jobExecutionId
-      Map<String, List<JournalRecord>> recordsByJobId = journalRecords.stream()
-        .collect(Collectors.groupingBy(JournalRecord::getJobExecutionId));
-
-      // Create the base SQL query
-      String query = format(INSERT_SQL, convertToPsqlStandard(tenantId), JOURNAL_RECORDS_TABLE);
-
-      // Process groups sequentially using recursive helper
-      List<RowSet<Row>> allResults = new ArrayList<>();
-      processNextGroup(new ArrayList<>(recordsByJobId.entrySet()), 0, query, tenantId, allResults, promise);
-
+      JsonObject[] records = journalRecords.stream()
+        .map(r -> new JsonObject()
+          .put("id", r.getId())
+          .put("job_execution_id", r.getJobExecutionId() != null ? r.getJobExecutionId() : null)
+          .put("source_id", r.getSourceId() != null ? r.getSourceId() : null)
+          .put("entity_type", r.getEntityType())
+          .put("entity_id", r.getEntityId())
+          .put("entity_hrid", r.getEntityHrId())
+          .put("action_type", r.getActionType())
+          .put("action_status", r.getActionStatus())
+          .put("action_date", r.getActionDate())
+          .put("source_record_order", r.getSourceRecordOrder())
+          .put("error", r.getError())
+          .put("title", r.getTitle())
+          .put("instance_id", r.getInstanceId())
+          .put("holdings_id", r.getHoldingsId())
+          .put("order_id", r.getOrderId())
+          .put("permanent_location_id", r.getPermanentLocationId())
+          .put("tenant_id", r.getTenantId()))
+        .toArray(JsonObject[]::new);
+
+      Tuple tuple = Tuple.tuple().addArrayOfJsonObject(records);
+      return pgClientFactory.createInstance(tenantId).execute("Select insert_journal_records($1::jsonb[])", tuple)
```

Comment:
we typically include schema name right?

Reply:
for the functions we dont define it 
"SELECT * FROM get_record_processing_log('%s', '%s')";